added 3D Preview of the models

TaylorMouse opened the pull request.

Commit:
added 3D Preview of the models

ZingBallyhoo commented:
no garuntees we will take this, but if we were to we would first need

all style/reformat changes undone. just causes massive clutter
obj code moved out of teModelChunk_RenderMesh.cs. owmdl code is in its own file and this should be too
delete teTriangle and just make a basic triangle type internal to obj writer
obj files shouldn't be written to disk for tankview preview

maybe other devs can come up with more

TaylorMouse commented:
Oh, ok, I thought the comparison of git spoke for itself but here are the changes:

So to be able to render in 3D I used a nuget package called HelixToolkit & HelixToolkit.Core.Wpf. Which I referenced in the PreviewDataModel.xaml ( the one that said "Soon!" )
So I think that is the correct place.
Since this is for quick viewing I converted it to obj first, cuz that is a format the Helix3DViewport natively understands as a format.

In the MainWindow.xaml I added a button that allows the conversion of the selected objects to export to obj as well.

Since I didn't find a reference to a Tris or triangle I used the naming convention you guys use, hence the teTriangle.

I also made the gridsplitter more obvious cuz I found that it was confusing from a UX point of view, but you can always undo that.

In the GUID Collection I need to pass in the selected object or item ( value ) 

So I don't really understand the "clutter" you are refering to .

ZingBallyhoo commented:
![image](https://user-images.githubusercontent.com/13888932/205169464-071c3646-e459-4659-9a6b-230ead35dbb2.png)
i mean dis. there is no code change here, just style

TaylorMouse commented:
Oh, sry about that, I sometimes press <ctrl> K + D for auto formatting the code, I might have pressed that by accident :/
